webapp/src/components/data/dataset_input/FileSelection.vue

```diff
@@ -90,76 +94,161 @@
   acceptFiles?: string[],
   isMultiple?: boolean,
-  infoText?: boolean
+  infoText?: boolean,
+  expectCsvMapping?: boolean
 }
+const props = withDefaults(defineProps<Props>(), {
+  isDirectory: false,
+  isMultiple: true,
+  acceptFiles: () => ['*'],
+  infoText: false,
+  label: undefined,
+  expectCsvMapping:false
+})
+
 const emit = defineEmits<Emits>()
 const selectedFiles = ref<FileList>()
 
 const uploadFile = ref<HTMLInputElement>()
 const uploadDirectory = ref<HTMLInputElement>()
 
-withDefaults(defineProps<Props>(), {
-  isDirectory: false,
-  isMultiple: true,
-  acceptFiles: () => ['*'],
-  infoText: false
+const hideConnectField = ref(false)
+
+const fileType = computed(() => {
+  if (props.expectCsvMapping === true || props.task.trainingInformation.dataType === 'tabular') {
+    return 'CSV'
+  } 
+  return props.task.trainingInformation.dataType === 'image' ? 'image' : 'file'
 })
+const requireLabels = computed(
+  () => props.task.trainingInformation.LABEL_LIST !== undefined
+)
 
-const clearFiles = () => {
-  emit('clear')
-  selectedFiles.value = undefined
-}
 const submitFiles = () => {
   if (uploadFile.value === undefined) return
   const files = uploadFile.value.files
   if (files === null) return
-
-  emit('input', files)
-  selectedFiles.value = files
+  addFiles(files)
 }
 const submitDirectory = () => {
   if (uploadDirectory.value === undefined) return
   const files = uploadDirectory.value.files
   if (files === null) return
-
-  emit('input', files)
-  selectedFiles.value = files
+  addFiles(files)
 }
+
 const dragFiles = (e: DragEvent) => {
   if (e.dataTransfer === null) return
-
   e.dataTransfer.dropEffect = 'copy'
-  emit('input', e.dataTransfer.files)
-  selectedFiles.value = e.dataTransfer.files
+  const files = e.dataTransfer.files
+  addFiles(files)
 }
+
+const addFiles = (files: FileList) => {
+  if (props.expectCsvMapping === true) {
+    readCsvImageMapping(files)
+    return
+  }
+  try{
+    const filesArray = Array.from(files)
+    // If the task is an image task with labels specified
+    if (props.task.trainingInformation.dataType === 'image' && requireLabels.value) {
+      if (props.label !== undefined) {
+        props.datasetBuilder.addFiles(filesArray, props.label)
+      } else {
+        if (props.csvRows === undefined) {
+          throw new Error('adding files but no CSV rows defined')
+        }
+        props.csvRows.forEach(row => {
+          // Match the selected files with the csv file names and label
+          const imageFile = filesArray.find(file => row.filename === file.name.split('.').slice(0, -1).join('.'))
+          if (imageFile === undefined) {
+            toaster.error("Some images listed in the CSV file are missing, make sure the CSV filenames don't include file extensions.")
+            throw new Error("Image not found in the CSV file")
+          } else if (imageFile !== undefined) {
+            props.datasetBuilder.addFiles([imageFile], row.label)
+          }
```

Comment:
unecessary if nesting, as the other branch throws. the flatter, the more readable it is as it avoid keeping more context while reading it.
```suggestion
          }
          props.datasetBuilder.addFiles([imageFile], row.label)
```